### ISSUE TITLE
508-defect-2 [SCREENREADER]: Map area SHOULD be hidden from screen readers, in the current state

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#10784]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Recommendation is to wrap the map area in a div with `aria-hidden="true" role="img"` to hide it from screen readers. It currently reads as a fast reading of letters, not even indicating that they are links. In a previous discussion with the VA 508 Office, they were okay with hiding the map from screen readers.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to understand the content of the page without confusing and irrelevant to me data.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: production

## Steps to Recreate

1. Enter https://www.va.gov/find-locations in browser
2. Enter a valid value in the first form input field and submit
3. Start screen reading device
4. Read through the page
5. Verify that the map pins read aloud as A B C D E F G I — with no interaction or context conveyed

## Possible Fixes (optional)

Consider wrapping the map area in a container with `aria-hidden="true" role="img"` to suppress the area for screen readers.

**Comment by 1Copenut** [comment]
@jenstrickland I concur, hiding the map is an okay solution, but I'd like it to be a temporary one as much as possible. That hinges on a lot of upstream code from Mapbox, so I understand that may be out of our control. That said, having a well-formatted and structured map that can be engaged by users with or without assistive tech is the ultimate goal.

Hiding the map from assistive tech while it's less than ideal seems the best decision for the moment.

**Comment by jenstrickland** [comment]
In my course with Marcy Sutton over the past few weeks, we covered making maps accessible. There were suggestions — such as the link provided on the epic containing this issue — yet, these issues primarily originate from Mapbox and Leaflet. If we're required to use this vendor product, I'm not sure what options we have. 

There is a work in progress by @lildziner to reevaluate the page and search results design, which I hope will render a more accessible solution, and allow removing the temporary patch of hiding the map from assistive tech. 🤞

**Comment by jenstrickland** [comment]
Good use case thinking. A preferred solution would be if it is visible / available for all, opposed to screen reader only text.  The challenge is that the map **should** be 508-compliant and accessible, but we aren't able to do that at this time due to the third-party tool limitations. 

If the map was initially hidden, and available only if a user selects "Show map" then all users would be aware of the map. Once the map is revealed, we may need to figure out a message until the map utility is accessible. @1Copenut do you have any thoughts on this?

**Comment by 1Copenut** [comment]
I'm on the fence with this one. Is the current best strategy to hide the map in favor of search results, and only show the map on user interaction like clicking a button `Show map` ? If we go that route, I would concur on having a message about the map being on the roadmap for updates. Not perfect, but acknowledging the current status.

**Comment by mmiddaugh** [comment]
### Summary of conversation with Jennifer and Trevor

- The ideal solution may not be possible with Mapbox. The most elegant workaround will require significant research, design and engineering, making implementation unlikely in the near future due to competing priorities. 
- However, we can immediately improve the user experience by hiding the map from assistive tech and providing a screenreader explanation.
- Any harm resulting from this approach is significantly less than the harm created in the current experience. 

### Plan: 

1. Wrap the map area (including attribution area) in a div with `aria-hidden="true" role="img"` to hide it from screen readers.
2. Replace text which would have introduced map content to screenreader with explanation that the map has been suppressed from screen readers. This will smooth the dissonance for low vision users who are aware the map is present but may be confused by the lack of screenreader presentation. 

> As we work through technical constraints to enhance map functionality, the visual presentation of search results is suppressed for screenreaders. 

(will seek input from content experts on this verbiage)

**Comment by peggygannon** [comment]
Hi, @mmiddaugh Based thread with Trevor and Jennifer, it sounds like we need a line of content explaining that the map is being suppressed from screenreaders? 

If so, then 
--does this map have existing content that sreeenreaders read currently? 
--Can I get a screenshot or visual of what the map looks like?  It's helpful for me to see what I'm writing to.

**Comment by mmiddaugh** [comment]
@peggygannon You are correct....we need a line of content explaining the map is suppressed from screenreaders. 
Right now, the map/icons are presented as a jarring wall of words which doesn't support understanding. The attribution information at the bottom of the map is also presented by the screenreader. 
Fortunately, the facility information is presented in the list so the map isn't the only way to understand the results. 
<details>
<summary> You can see the map at https://www.va.gov/find-locations/ or in the screenshot below. </summary> 

![screencapture-va-gov-find-locations-2020-09-17-14_01_01](https://user-images.githubusercontent.com/55411834/93522819-4d8a4180-f8ef-11ea-888f-978f5862da4f.png)

</details>

**Comment by peggygannon** [comment]
@mmiddaugh ok, thank you. Here's recommended content. Do we need to provide the users any reasons why or is this 1 line enough? 

**Please note:** The map has been hidden from screen reader devices.

**Comment by mmiddaugh** [comment]
@jenstrickland @1Copenut does Peggy's proposed language work for you?

**Comment by 1Copenut** [comment]
@peggygannon && @mmiddaugh I'd lean more toward offering a reason why the map has been hidden, if we can offer it in 1-2 additional sentences.

**Comment by peggygannon** [comment]
thanks, @1Copenut, is this an accurate (better) explanation?  


**Please note:** There is a map that's hidden from screen reader devices. This type of assistive technology has a hard time reading the map and its symbols.  To see the map, xxxcontent tktktktk { @mmiddaugh what is the option if caregivers want to  to open the map?} Please let me know and we can add that to this note.

**Comment by jenstrickland** [comment]
@peggygannon 

The map will be open and visible for anyone who has sight. It just won't be read by screen readers.

At present, a screen reader just reads **ABCDEFGHIJKLMNOPQ** — and it sounds like it is being shouted. There's no information that communicates what a sighted user gets from the map, with regard to cross-streets or geography. 

We would want to explain that we're working on improving the screen reader experience of the map, and suppressing it for now because it is not useful.

**Comment by mmiddaugh** [comment]
**Please note:** Due to technical constraints, the map has been hidden from screen reader devices.

@1Copenut @peggygannon Thoughts?

**Comment by peggygannon** [comment]
hello, @mmiddaugh, @1Copenut @jenstrickland OK, what about this? (sometimes it takes a village....)

**Please note:** Due to technical limitations, the map has been hidden from screen reader devices. This type of assistive technology has a hard time reading the map and its symbols. We’re working to fix this.

**Comment by jenstrickland** [comment]
Rather than faulting the assistive tech, can we own that maps online aren't screen reader friendly? It's not the fault of the screen readers. It's that designers and developers don't build maps with assistive tech users in mind. 

Perhaps:
Please note: Due to technical limitations, the map has been hidden from screen reader devices. We're working to deliver an enhanced screen reader experience.

**Comment by peggygannon** [comment]
Hi, @jenstrickland thank you for the additional feedback. that ^^ works for me.

**Comment by brangi** [comment]
@jenstrickland @1Copenut, I am running into some issues trying to accomplish this. Please refer to my comments here:

https://github.com/department-of-veterans-affairs/vets-website/pull/14397#issuecomment-700933628
https://github.com/department-of-veterans-affairs/vets-website/pull/14397#issuecomment-700960853

**Comment by mmiddaugh** [comment]
@jenstrickland this is ready for your validation when you are available.

**Comment by jenstrickland** [comment]
@mmiddaugh The paragraph was added **after** the map, at the bottom of the map container. It should be the first item inside the map container.

![Screen Shot 2020-10-01 at 6 29 15 PM](https://user-images.githubusercontent.com/57469/94869827-1e59f100-0414-11eb-9b3a-5657367dff09.png)

**Comment by mmiddaugh** [comment]
The change has been made. Paragraph is now the first item within the container. 
![image](https://user-images.githubusercontent.com/55411834/95257160-24b7e680-07e1-11eb-9368-07ecbad0c5a1.png)
